code/modules/mob/typing_indicator.dm

```diff
@@ -0,0 +1,44 @@
+/// state = overlay/image/object/type/whatever add_overlay will accept
+GLOBAL_LIST_EMPTY(typing_indicator_overlays)
+
+/// Fetches the typing indicator we'll use from GLOB.typing_indicator_overlays
+/mob/proc/get_indicator_overlay(state)
+	. = GLOB.typing_indicator_overlays[state]
+	if(.)
+		return
+	// doesn't exist, make it and cache it
+	if(ispath(state))
+		. = GLOB.typing_indicator_overlays[state] = state
+	// We only support paths for now because anything else isn't necessary yet.
+
+/// Gets the state we will use for typing indicators. Defaults to src.typing_indicator_state
+/mob/proc/get_typing_indicator_icon_state()
+	return typing_indicator_state
+
+/*!
```

Comment:
IIRC, the proper doc format uses two asterisks and no exclamation point, as well as the following lines  of the commentary having two spaces at the start instead of one, but w/e.